src/dbnode/storage/series/reader.go

```diff
@@ -117,6 +117,18 @@ func (r Reader) readersWithBlocksMapAndBuffer(
 
 	first, last := alignedStart, alignedEnd
 	for blockAt := first; !blockAt.After(last); blockAt = blockAt.Add(size) {
+		// resultsBlock holds the results from one block. The flow is:
+		// 1) Look in the cache for metrics for a block.
+		// 2) If there is nothing in the cache, try getting metrics from disk.
+		// 3) Regardless of (1) or (2), look for metrics in the series buffer.
+		//
+		// It is important to look for data in the series buffer one block at
+		// a time within this loop so that the returned results contain data
+		// from blocks in chronological order. Failure to do this will result
+		// in an out of order error in the MultiReaderIterator on query.
+		resultsBlock := []xio.BlockReader{}
```

Comment:
Just to avoid this allocating if not necessary, can we use `var resultsBlock []xio.BlockReader`?